Main solution E2E

android-dev-lxl opened the pull request.

johnshea reviewed (changes requested):
The UI on API 33 needs to be adjusted. It changed and no longer matches the pre e2e UI.

<img width="802" alt="Screenshot 2023-08-21 at 9 54 51 AM" src="https://github.com/google-developer-training/basic-android-kotlin-compose-training-tip-calculator/assets/7107817/94c9f4aa-ef1c-4517-87df-70cf73c7f6b0">

johnshea commented on build.gradle.kts:
```diff
@@ -16,12 +16,12 @@
 buildscript {
     extra.apply {
-        set("compose_compiler_version", "1.4.7")
+        set("compose_compiler_version", "1.5.1")
```
Update `kotlinCompilerExtensionVersion`

johnshea commented on build.gradle.kts:
```diff
@@ -16,12 +16,12 @@
-    id("com.android.application") version "8.0.2" apply false
-    id("com.android.library") version "8.0.2" apply false
-    id("org.jetbrains.kotlin.android") version "1.8.21" apply false
+    id("com.android.application") version "8.1.0" apply false
```
Update AGP

johnshea commented on app/src/main/java/com/example/tiptime/MainActivity.kt:
```diff
@@ -83,19 +83,18 @@ fun TipTimeLayout() {
             text = stringResource(R.string.calculate_tip),
             modifier = Modifier
-                .padding(bottom = 16.dp)
+                .padding(bottom = 16.dp, top = 40.dp)
```
Could we add 40.dp to the dimens.xml? It is used in multiple spots.

android-dev-lxl replied:
We do not have `dimens.xml` for this project.

johnshea commented on app/src/main/java/com/example/tiptime/MainActivity.kt:
Could we add 32.dp to the dimens.xml? Used in multiple spots.

android-dev-lxl replied:
Same as above, me and SME decided that is too early to introduce dimens.xml

johnshea reviewed (changes requested):
Navigation bar is not translucent (Pixel 6 API 33) but should be.

<img width="548" alt="Screenshot 2023-08-25 at 10 59 27 AM" src="https://github.com/google-developer-training/basic-android-kotlin-compose-training-tip-calculator/assets/7107817/df874020-1e83-41cb-97a1-47f00e898ae5">

API 24 - Navigation bar is not translucent but should be. (Note - I added additional `Text` to test.)

<img width="553" alt="Screenshot 2023-08-25 at 11 29 26 AM" src="https://github.com/google-developer-training/basic-android-kotlin-compose-training-tip-calculator/assets/7107817/0715d3fc-40ca-4ae1-9cc4-7cb1cb4f1c25">

API 26 - Navigation bar is not translucent but should be. (Note - I added additional `Text` to test.)
<img width="552" alt="Screenshot 2023-08-25 at 11 29 58 AM" src="https://github.com/google-developer-training/basic-android-kotlin-compose-training-tip-calculator/assets/7107817/b6019726-9e2e-4306-b21c-29dd65c0e7e9">

johnshea reviewed (changes requested):
Issue 1

UI components are not positioned the same for "System Navigation - Gesture" vs "System Navigation - 3 button" (Pixel 6 API 33)

Note - Red lines drawn to show how UI components are not positioned the same. Lines are not part of the actual UI.

<img width="1022" alt="Screenshot 2023-08-30 at 6 18 09 PM" src="https://github.com/google-developer-training/basic-android-kotlin-compose-training-tip-calculator/assets/7107817/cab34a8d-493f-43fd-a57a-661f47a2c2e0">

-----

Issue 2

UI scrolls vertically through the status bar.

Is this the intended behavior?

<img width="543" alt="Screenshot 2023-08-30 at 6 29 07 PM" src="https://github.com/google-developer-training/basic-android-kotlin-compose-training-tip-calculator/assets/7107817/1387a20a-03fc-44df-a14a-3d73cbc7f2a7">

android-dev-lxl commented:
> Navigation bar is not translucent (Pixel 6 API 33) but should be.
> 
> <img alt="Screenshot 2023-08-25 at 10 59 27 AM" width="548" src="https://user-images.githubusercontent.com/7107817/263311105-df874020-1e83-41cb-97a1-47f00e898ae5.png">
> API 24 - Navigation bar is not translucent but should be. (Note - I added additional `Text` to test.)
> 
> <img alt="Screenshot 2023-08-25 at 11 29 26 AM" width="553" src="https://user-images.githubusercontent.com/7107817/263315070-0715d3fc-40ca-4ae1-9cc4-7cb1cb4f1c25.png">
> API 26 - Navigation bar is not translucent but should be. (Note - I added additional `Text` to test.) <img alt="Screenshot 2023-08-25 at 11 29 58 AM" width="552" src="https://user-images.githubusercontent.com/7107817/263315178-b6019726-9e2e-4306-b21c-29dd65c0e7e9.png">



Issue 1 - Is not an issue and is expected. We can chat about this, if needed. 
Issue 2 - Fixed

android-dev-lxl commented:
> Issue 1
> 
> UI components are not positioned the same for "System Navigation - Gesture" vs "System Navigation - 3 button" (Pixel 6 API 33)
> 
> Note - Red lines drawn to show how UI components are not positioned the same. Lines are not part of the actual UI.
> 
> <img alt="Screenshot 2023-08-30 at 6 18 09 PM" width="1022" src="https://user-images.githubusercontent.com/7107817/264486872-cab34a8d-493f-43fd-a57a-661f47a2c2e0.png">
> Issue 2
> 
> UI scrolls vertically through the status bar.
> 
> Is this the intended behavior?
> 
> <img alt="Screenshot 2023-08-30 at 6 29 07 PM" width="543" src="https://user-images.githubusercontent.com/7107817/264488044-1387a20a-03fc-44df-a14a-3d73cbc7f2a7.png">

Issue 1 - Is not an issue and is expected. We can chat about this, if needed.
Issue 2 - Fixed

johnshea reviewed (approved):
lgtm

one nit - remove comment in MainActivity.kt as the code it refers to was removed.

johnshea commented on app/src/main/java/com/example/tiptime/ui/theme/Theme.kt:
```diff
@@ -17,6 +17,7 @@ package com.example.tiptime.ui.theme
 
 import android.app.Activity
```
Remove unused imports

johnshea reviewed (approved):
Approved but unused imports need to be removed.
